src/components/base/ucc_base_iface.h

```diff
@@ -68,6 +68,14 @@ typedef struct ucc_base_context_iface {
 
 typedef struct ucc_base_team_params {
     ucc_team_params_t params;
+    int               scope; /* Scope that allocates the team. When TL team is created
+                                the scope would be a CL_TYPE. This provides a separation
+                                of teams created from different CLs with the same TL_TYPE */
+    int               scope_id; /* The id of the base_team in the specified scope. Use case:
+                                   a single CL team (e.g. basic) creates multiple TL teams
+                                   of the same type (e.g. several tl_ucp teams). In this
+                                   case CL would give those teams different scope_id. */
+    uint32_t          rank; /* Rank of a calling process in the TL/CL team */
```

Comment:
We use ep to mean a participant of the team externally. Do we want to call it team_ep instead of rank? Moving away from MPI kind of understand and constraints. I’m assuming this rank is unique to the team but not necessarily to the entire job.

Reply:
it is internal interface, and this value is exactly "rank", not endpoint. This has nothing to do with MPI. It is a uniq process identifier within a team (not job) but it has the property: it is always contig and in the rang [0, team_size). It is just convenient to call it rank. It is part of internal interface because we always need such a value to implement any meaningful algorithm (except for some dummy algs where everybody just send everything to some random endpoint). Externally EP is supposed to be used for alternative way of team creations but to write algorithms we need "rank". If user does not give us one, we need to generate it in the "core" and then pass down to cl/tl.

Reply:
Can you please add "It is a uniq process identifier within a team (not job) but it has the property: it is always contig and in the rang [0, team_size). " to the documentation? It might be useful for readers.

Reply:
manju, do you mean internal comment in the code? it is not part of the public api.

Reply:
Internal. Right next to the structure.

Reply:
added